Restructure architecture (#118)
openclimatedata/openscm: docs/usage.rst

OpenSCM defines and provides three interfaces for its users.

Its **Core** interface is targeted to users who want to include
OpenSCM in their model, e.g. integrated assessment modellers. This
interface provides the basic functionality necessary to run all SCMs
included in OpenSCM. It includes functions for getting and setting
parameters as well as to run and reset the model.

The **Pythonic** interface adds several convenience functions on top
of the core interface. It allows for reading and writing parameters

> from and to a standardized and other file formats, including whole

Reviewer: ```suggestion
from and to standardized and other file formats, including whole
```

> scenario definitions. Also, it provides functions for running

Reviewer: ```suggestion
scenario definitions. It also provides functions for running
```

> ensembles of model runs and doing stochastic analyses and model

Reviewer: ```suggestion
ensembles of model runs.
```

Reviewer: The term stochastic is misleading and I think we should get rid of it

> tuning.

Reviewer: ```suggestion

```

    ``Emissions`` -> ``CO2`` -> ``Industrial``.

In the core API parameters are expected to be identified by tuples of
strings describing their position in the hierarchy, i.e. in this

> example ``("Emissions", "CO2", "Industrial")``. The ScmDataFrame API

Reviewer: ```suggestion
example ``("Emissions", "CO2", "Industrial")``. 
```

Delete the whole ScmDataFrame API sentence. We'll explain how to access data when we write the ScmDataFrame API docs

Setting input parameters
************************

> In the core API parameters are get and set through of subclasses of

Reviewer: ```suggestion
In the core API parameters are get and set through subclasses of
```

:class:`ParameterView`. While the values of the parameters are stored
internaly, a :class:`ParameterView` provides an (always up-to-date)

> "view" of the corresponding parameter giving the parameter values in a

Reviewer: ```suggestion
"view" of the corresponding parameter and will always return the parameter values in a
```
